lib/src/utils/feature_flags_manager.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:instabug_flutter/src/generated/instabug.api.g.dart';
+import 'package:instabug_flutter/src/utils/ibg_build_info.dart';
+import 'package:meta/meta.dart';
+
+typedef OnW3CFeatureFlagChange = void Function(
+  bool isW3cExternalTraceIDEnabled,
+  bool isW3cExternalGeneratedHeaderEnabled,
+  bool isW3cCaughtHeaderEnabled,
+);
+
+class FeatureFlagsManager implements FeatureFlagsFlutterApi {
```

Comment:
Let's make this class a singleton instead of using static methods since we can't mock them if needed.